File: AndroidBasicFundamental/ImageGlideApp/app/src/main/java/com/algokelvin/glide/DataAdapter.java
package com.algokelvin.glide;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.algokelvin.glide.model.ImageUrl;
import com.bumptech.glide.Glide;

import java.util.ArrayList;

public class DataAdapter extends RecyclerView.Adapter<DataAdapter.ViewHolder> {
    private final ArrayList<ImageUrl> imageUrls;
    private final Context context;

    public DataAdapter(Context context, ArrayList<ImageUrl> imageUrls) {
        this.context = context;
        this.imageUrls = imageUrls;
    }

    @NonNull
    @org.jetbrains.annotations.NotNull
    @Override
    public DataAdapter.ViewHolder onCreateViewHolder(@NonNull @org.jetbrains.annotations.NotNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.image_layout, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull @org.jetbrains.annotations.NotNull DataAdapter.ViewHolder holder, int position) {
        Glide.with(context).load(imageUrls.get(position).getImageUrl()).into(holder.img);
    }

    @Override
    public int getItemCount() {
        return 0;
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        ImageView img;

        public ViewHolder(View view) {
            super(view);
            img = view.findViewById(R.id.imageView);
        }
    }
}

File: AndroidExpert/MovieApp/app/src/main/java/com/algokelvin/moviecatalog/ui/adapter/DataAdapter.kt
package com.algokelvin.moviecatalog.ui.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView

class DataAdapter(
    private val count: Int,
    private val layout: Int,
    private val viewClick : (View, Int) -> Unit
): RecyclerView.Adapter<DataAdapter.DataViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): DataViewHolder =
        DataViewHolder(LayoutInflater.from(parent.context).inflate(layout, parent, false))

    override fun onBindViewHolder(holder: DataViewHolder, position: Int) {
        viewClick(holder.itemView, position)
    }

    override fun getItemCount(): Int = count

    inner class DataViewHolder(itemView: View): RecyclerView.ViewHolder(itemView)
}